### PR TITLE
Uses COALESCE_LINKS instead of COALESCE_ARRAYS

### PR DESCRIPTION
**Pull request by sventorben**
Fixes HalBuilder/halbuilder-json#18

### CODE REVIEW

File: src/test/java/com/theoryinpractise/halbuilder/json/TestIssue18.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    Representation representation = representationFactory.newRepresentation("http://localhost/self");
+    representation.withRepresentation("items", embedded);
+
+    String json = representation.toString(RepresentationFactory.HAL_JSON);
+    ObjectNode objectNode = (ObjectNode) new ObjectMapper().readTree(json);
+
+    assertFalse(objectNode.get("_embedded").get("items").get(0).get("_links").get("applicants").isArray());
+  }
+
+}
```
Comment (talios) [inline src/test/java/com/theoryinpractise/halbuilder/json/TestIssue18.java]: nitpick - can we add the newline.

**Comment by sventorben** [comment]
Is there a chance for a maven release?

**Comment by talios** [comment]
Yep - hopefully I'll get to to is this weekend.  Just had a super busy period at work and concert shoots. Currently the tests fail due to now dropping down to an object rather than a link.  So they need a bit of a tweak ( either the test, or the test data ).  Otherwise I'd have released on the spot.

**Comment by talios** [comment]
4.0.5 released after fixing embedded links and tests.  Sorry for the delays.

**Comment by sventorben** [comment]
Cool! Thx.

**Comment by talios** [comment]
Actually, looks like I was wrong about my own code :( Spending too much time with the newer API.  COALESCE_LINKS is actually for combining multiple links to a common href to a single space-separated rel ( i.e. "self edit" or "parent customer". Whilst COALESCE_ARRAYS was for coalescing... the arrays.

This explains why none of the tests worked.

I think, I might have to revert both the PR commit and my subsequent 'test fix' commit and re-release with the original behaviour.

**Comment by sventorben** [comment]
I was just trying to integrate the new version into our production code: Some of our tests fail as well.

Initially I was looking for a way to achieve what COALESCE_ARRAYS should do, but for links only. We do not want arrays from our beans to coalesce when only a single element is present. 
In addition, we do not want our links to be single space-separated, but get rid of the array if there is a single href for a given link. 
Maybe a "COALESCE_LINK_ARRAYS" would be useful.

**Comment by talios** [comment]
Bean properties should NOT be dropping arrays to a single element. This is explicitly disabled by [`JsonRepresentationWriter`](https://github.com/HalBuilder/halbuilder-json/blob/support/halbuilder-json-4.x/src/main/java/com/theoryinpractise/halbuilder/json/JsonRepresentationWriter.java#L68).

What you describe is actually the original bevaiour, COALESCE_LINKS only did the space separated collapsing ( in halbuilder-core ) where COALESCE_ARRAYS was only so for link/embedded objects.

The 4.x series is basically in 'low maintenance' whilst I work ( slowly ) on the new Halbuilder 5 API which removes completely both of these flags ( flags in general ) whilst giving per-rel repetition control.

Can you provide an example of what failed for you? And how the representation factory is created?

**Pull request merged**